### PR TITLE
Revist FPC validation diagram

### PR DESCRIPTION
**Pull request by mbrandenburger**
**What this PR does / why we need it**:

This PR completes the entire FPC flow with the last missing step - the validation.

* replace MSP validtion with custom vscc plugin

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing changes and/or breaks backward compatability?**:

```

### CODE REVIEW

File: docs/design/fabric-v2+/fpc-validation.puml [hunk 1]
```diff
@@ -33,33 +33,11 @@ ref over User1, Orderer
...
 
-note right ECC3 #FFaa00
-    How can we express that we need an enclave here?
-    In particular, an enclave that has ECC_MR and also has a valid attestation?!
-end note
-
 ref over User1, Orderer
     register enclaves and key-dist
     see fpc-registration.puml and fpc-key-dist.puml
 end ref
```
Comment (g2flyer) [inline docs/design/fabric-v2+/fpc-validation.puml:39]: I cannot add a suggestion as it is not in your diff, but we should mention designated peer endorsement policy as 2 of 3 actually doesn't work with the plugin approach (or we also need deterministic signatures and alike ...)

Actually, might be worth mention this limitation, this is only for MVP and we have architecture via MSP to go beyond?

File: docs/design/fabric-v2+/fpc-validation.puml [hunk 2]
```diff
@@ -97,22 +75,32 @@ end group
...
+
+            group default validation
+                ECC1 -> ECC1 : perform default validation
+                ECC1 -> ECC1 : validate tx (client) signature
+                ECC1 -> ECC1 : MVCC check
+                ECC1 -> ECC1 : validate endorsements
+            end
+
+            group fpc-vscc
+                ECC1 -> ECC1 : enclave_id <- extract from transaction
```
Comment (g2flyer) [inline docs/design/fabric-v2+/fpc-validation.puml:92]: Don't we need here also first a parsing of the payload into actual fpc-and fpc metadata (signature, enclave id, ..)?  Extraction of id is only one part, you would need extracted signature (so it's not confused with normal endorsement signature).  Ideally, we use the same decomposition/terms as in fpc-cc-invocation ..

File: docs/design/fabric-v2+/fpc-validation.puml [hunk 3]
```diff
@@ -97,22 +75,32 @@ end group
...
+                create participant ERCC order 30
+                ECC1 -> ERCC++ : get_attestation(enclave_id)
+                return attestation
+                ECC1 -> ECC1 : check attestation matches enclave identity
```
Comment (g2flyer) [inline docs/design/fabric-v2+/fpc-validation.puml:96]: Why do we need the attestation?  we really just care here that enclave_id was registered for that chaincode?
```suggestion
                ECC1 -> ERCC++ : check_registration(cc_name, enclave_id)
                return attestation
                ECC1 -> ECC1 : true/false
```

File: docs/design/fabric-v2+/fpc-validation.puml [hunk 4]
```diff
@@ -97,22 +75,32 @@ end group
...
+                ECC1 -> ECC1 : verify enclave signature
```
Comment (g2flyer) [inline docs/design/fabric-v2+/fpc-validation.puml:97]: Worth mentioning what the signature is over? (i.e., symmetric to the generation in fpc-cc-invocation; obviously depends also on how you handled my comment above on parsing)

File: docs/design/fabric-v2+/fpc-validation.puml (hunk 1 above)
Comment (mbrandenburger) [reply docs/design/fabric-v2+/fpc-validation.puml:39]: Ok added a comment

File: docs/design/fabric-v2+/fpc-validation.puml (hunk 2 above)
Comment (mbrandenburger) [reply docs/design/fabric-v2+/fpc-validation.puml:92]: Updated the flow giving more details and sync with terms used in fpc-cc-invocation

File: docs/design/fabric-v2+/fpc-validation.puml (hunk 3 above)
Comment (mbrandenburger) [reply docs/design/fabric-v2+/fpc-validation.puml:96]: Oh good catch! I suggest the following. Let's introduce queryECCInfo and queryECCInfos. (this is similar what Lifecycle exposes for chaincode definitions). 
I update invocation diagram accordingly

File: docs/design/fabric-v2+/fpc-validation.puml [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+            group default validation
+                ECC1 -> ECC1 : perform default validation
+                ECC1 -> ECC1 : validate tx (client) signature
+                ECC1 -> ECC1 : MVCC check
+                ECC1 -> ECC1 : validate endorsements
+            end
+
+            group fpc-vscc
+                ECC1 -> ECC1 : <args, read/writeset, result> <- extract from transaction
```
Comment (g2flyer) [inline docs/design/fabric-v2+/fpc-validation.puml:96]: Folks might not know what fpc-vscc is
```suggestion
            group fpc-vscc
                note over ECC1
                     fpc-vscc is a validation plugin associated with each fpc chaincode.
                end note
                ECC1 -> ECC1 : <args, read/writeset, result> <- extract from transaction
```

File: docs/design/fabric-v2+/fpc-validation.puml [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+        see fpc-cc-invocation.puml
+    end ref
+    return proposal response
+
+    User1 -> ECC2 ++: send proposal
+    ref over ECC2
+        invoke chaincode enclave
+        see fpc-cc-invocation.puml
+    end ref
+    return
```
Comment (bvavala) [inline docs/design/fabric-v2+/fpc-validation.puml:60]: Given that this is a "designated peer" scenario, it is misleading to have multiple endorsers.

```suggestion
```

File: docs/design/fabric-v2+/fpc-validation.puml [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+            validation & commit
+        end ref
+        note over TLCC1
+            Ledger enclave performs the same
+            validation steps as the peer does.
+            This is necessary to let the ledger
+            enclave establish its own trusted
+            view of the ledger.
+        end note
+        |||
```
Comment (bvavala) [inline docs/design/fabric-v2+/fpc-validation.puml:144]: Missing integrity metadata...
```suggestion
        end note
        ECC1 -> ECC1 : compute and store integrity-metadata for the invoked FPC chaincode's writeset
        |||
```

**Comment by g2flyer** [comment]
> two small clarification suggestions

Yep, good finds! I've accepted them as an independent approval, so you or i can squash-merge.  I realize now, though, that i should have added a `Signed-off-by: ..`  as marcus has done ..  and it would have removed the need to overrule the DCO check ...

**Pull request merged**